Refactor syntax unicode datetime

livecodefraser opened the pull request.

Commit:
Change MCDateTimeLocale to use StringRefs internally.

A number of the time/date routines have also been changed to deal in StringRefs. Additionally, g_english locale has been renamed to g_basic_locale in order to better reflect is usage and contents.

livecodesebastien commented on engine/src/date.cpp:
```diff
@@ -1348,21 +1314,28 @@ bool MCD_convert_from_datetime(MCExecPoint &p_context, Convert_form p_primary_to
+			t_buffer = t_new;
 		}
+		
+		p_context . setvalueref(t_buffer);
```
`t_buffer` should be released at the end of the function.

livecodesebastien commented:
Updating the `char*` to `MCStringRef` of `MCDateTime` also affects platform-specific files/functions:
- `cache_locale()` in sysunxdate.cpp
-  `windows_cache_locale()` in w32date.cpp
  which therefore need to be updated as well - Linux and Windows engine won't compile otherwise.
  Also, the update breaks the parsing: `put ("Wednesday, September 18 2013" is a date)` returns False in the new version, but used to return True.

Commit:
Fix an off-by-one error causing date parsing to fail

Number parsing would fail if it was the last item in the input string.

Commit:
Fix a reference leak in MCD_convert_from_datetime

Commit:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_datetime

Commit:
Fix to date/time handling for Unix systems

The Unix date/time routines were not fully updated previously.

Commit:
Fix date/time formatting for Win32

Win32 date/time formatting code was not yet updated to use the StringRef date/time functions.
